packages/rs-platform-value/src/types/bytes_20.rs

```diff
@@ -0,0 +1,201 @@
+use crate::string_encoding::Encoding;
+use crate::types::encoding_string_to_encoding;
+use crate::{string_encoding, Error, Value};
+use bincode::{Decode, Encode};
+use serde::de::Visitor;
+use serde::{Deserialize, Serialize};
+use std::fmt;
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Hash, Copy, Encode, Decode)]
+pub struct Bytes20(pub [u8; 20]);
+
+impl AsRef<[u8]> for Bytes20 {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl Bytes20 {
+    pub fn new(buffer: [u8; 20]) -> Self {
+        Bytes20(buffer)
+    }
+
+    pub fn from_vec(buffer: Vec<u8>) -> Result<Self, Error> {
+        let buffer: [u8; 20] = buffer.try_into().map_err(|_| {
+            Error::ByteLengthNot20BytesError("buffer was not 20 bytes long".to_string())
+        })?;
+        Ok(Bytes20::new(buffer))
+    }
+
+    pub fn as_slice(&self) -> &[u8] {
```

Comment:
BTW, you can use `Deref` trait, so any method that takes `&[u8]` can take `&Bytes20` as well:

```rust
use std::ops::Deref;

pub struct Bytes32([u8; 32]);

impl Deref for Bytes32 {
    type Target = [u8];

    fn deref(&self) -> &Self::Target {
        &self.0
    }
}

fn take_ref(data: &[u8]) -> &[u8] {
    data
}

fn example_main() {
    let bytes = Bytes32([0; 32]);

    take_ref(&bytes);
}
```